src/devtools/client/debugger/src/components/App.tsx

```diff
@@ -5,45 +5,51 @@
 import React, { Component, useEffect, useState, useRef } from "react";
 import PropTypes from "prop-types";
 
-import { connect } from "react-redux";
-import { features } from "../utils/prefs";
-import { prefs } from "../../../../../ui/utils/prefs";
+import { connect, ConnectedProps } from "react-redux";
+
+import type { UIState } from "ui/state";
 import actions from "../actions";
 import { setUnexpectedError } from "ui/actions/errors";
 import A11yIntention from "./A11yIntention";
-import { ShortcutsModal } from "./ShortcutsModal";
-import SplitBox from "devtools/client/shared/components/splitter/SplitBox";
-
-import {
-  getActiveSearch,
-  getPaneCollapse,
-  getQuickOpenEnabled,
-  getSelectedSource,
-} from "../selectors";
 
 import { getSelectedPanel } from "ui/reducers/layout";
-import { getToolboxLayout } from "ui/reducers/layout";
 import { useGetUserSettings } from "ui/hooks/settings";
 
 import KeyShortcuts from "devtools/client/shared/key-shortcuts";
 
 import { EditorPane } from "./Editor/EditorPane";
 
-class Debugger extends Component {
+const mapStateToProps = (state: UIState) => ({
```

Comment:
Hmmmmm I see we are pulling `connect` code *up* here. I think most of our files have this towards the bottom, and in fact I've actually seen folks pulling these *down* to match other places. I am used to seeing them at the bottom, but I don't care that much as long as we follow *something* consistently. (And obviously the number of connect statements should be going down over time because hooks exist).

Reply:
Yeah, personally I've always put the `connect` call itself at the bottom, but the `mapState/mapDispatch` definitions _above_ the component, so you can read downwards and see what props it's getting.

Beyond that, there's a couple issues here:

- The `ConnectedProps<T>` usage requires splitting the `connect` call into `const connector = connect(mapState, mapDispatch)`, and then inferring `type PropsFromRedux = ConnectedProps<typeof connector>`
- _Technically_ you can define the props types down at the bottom of the file and use them earlier, but that does feel really awkward to me. I would expect to see the props types defined first _before_ they're used in the component